base/gfspvgmgr/virtual_group_manager.go

```diff
@@ -785,30 +787,39 @@ func (checker *HealthChecker) checkSPHealth(sp *sptypes.StorageProvider) bool {
 		Transport: &http.Transport{
 			TLSClientConfig: &tls.Config{MinVersion: tls.VersionTLS12},
 		},
-		Timeout: defaultSPCheckTimeout * time.Second,
+		Timeout: defaultSPCheckTimeout,
 	}
 
 	// Create an HTTP request to test the validity of the endpoint
 	urlToCheck := fmt.Sprintf("%s%s", endpoint, httpStatusPath)
-	req, err := http.NewRequestWithContext(ctxTimeout, http.MethodGet, urlToCheck, nil)
-	if err != nil {
-		return false
-	}
+	for attempt := 0; attempt < defaultSPHealthCheckerMaxRetries; attempt++ {
+		start := time.Now()
+		req, err := http.NewRequestWithContext(ctxTimeout, http.MethodGet, urlToCheck, nil)
+		if err != nil {
+			log.CtxErrorw(context.Background(), "failed to create request", "sp", sp, "error", err)
+			return false
+		}
 
-	resp, err := client.Do(req)
-	if err != nil {
-		log.CtxErrorw(context.Background(), "failed to connect to sp", "sp", sp, "error", err)
-		return false
-	}
-	defer resp.Body.Close()
+		resp, err := client.Do(req)
+		duration := time.Since(start)
+		if err != nil {
+			log.CtxErrorw(context.Background(), "failed to connect to sp", "sp", sp, "error", err, "duration", duration)
+			time.Sleep(defaultSPHealthCheckerRetryInterval)
+			continue
+		}
+		defer resp.Body.Close()
 
-	if resp.StatusCode != http.StatusOK {
-		log.CtxErrorw(context.Background(), "failed to check sp healthy", "sp", sp, "http_status_code", resp.StatusCode, "resp_body", resp.Body)
-		return false
+		if resp.StatusCode == http.StatusOK {
+			log.CtxInfow(context.Background(), "succeed to check the sp healthy", "sp", sp, "duration", duration)
```

Comment:
can we put a metric for the duration so that later we can setup monitor?

Reply:
added